src/AutoMapper/Mappers/DynamicMappers.cs

```diff
@@ -21,6 +21,7 @@ public class FromDynamicMapper : IObjectMapExpression
                 destination = (TDestination) (!context.ConfigurationProvider.AllowNullDestinationValues
```

Comment:
@jbogard @lbargaoanu This perhaps?  It's pretty ugly and probably can be optimized more.